m4/openssl.m4

```diff
@@ -220,8 +223,8 @@ int main(void) { DTLSv1_method(); }
 AC_MSG_CHECKING([if programs using OpenSSL functions will link])
 AC_LINK_IFELSE(
 	[AC_LANG_SOURCE([[
-#include <openssl/evp.h>
-int main(void) { SSLeay_add_all_algorithms(); }
+#include <openssl/crypto.h>
+int main(void) { OpenSSL_version_num(); }
```

Comment:
Do we need explicit `return` or `exit()`?

Do we need to change the following similar `AC_LINK_IFELSE()` too?

Reply:
`main` has a special exception in the standard which I don't expect to ever change, but I can change it to avoid issues with e.g. LTO (we've seen issues in the past where such unused calls can be optimised out, see https://gitlab.kitware.com/cmake/cmake/-/issues/26103).

I'll check the other `AC_LINK_IFELSE`.

Reply:
Done. Sorry for the wait.